core/modules/filters/kin.js

```diff
@@ -0,0 +1,96 @@
+/*\
+title: $:/core/modules/filters/kin.js
+type: application/javascript
+module-type: filteroperator
+
+Finds out where a tiddler originates from and what other tiddlers originate from it
+
+\*/
+(function() {
+
+	/*jslint node: true, browser: true */
+	/*global $tw: true */
+	"use strict";
+
+	function collectTitlesRecursively(baseTiddler,baseTitle,options) {
```

Comment:
A big issue to consider with this operator is efficiency. You'll notice that the wiki object has some features for caching expensive computations, and that these are used for most of the tag-related operators. I think we need to do something similar here to make this moderately efficient.